server/src/main/java/org/elasticsearch/plugins/RepositoryPlugin.java

```diff
@@ -67,7 +67,7 @@ default Map<String, Repository.Factory> getInternalRepositories(
      *
      * returns null if no check is provided
      */
-    default BiConsumer<Snapshot, Version> addPreRestoreVersionCheck() {
+    default BiConsumer<Snapshot, IndexVersion> addPreRestoreVersionCheck() {
```

Comment:
This is a plugin API change

Reply:
As far as I know, the point of this API is to check Lucene version, and that's its only use. Since the old plugin API is known to be unstable, I think it's okay to make this change.